client/src/components/pages/CreatePostPage/Event/index.js

```diff
@@ -0,0 +1,243 @@
+
+const InputGroup = Input.Group;
+
+class EventForm extends React.Component {
+  handleSubmit = e => {
+    e.preventDefault();
+    this.props.form.validateFieldsAndScroll((err, values) => {
+      if (!err) {
+        // for fetch
+      }
+    });
+  };
+
+  render() {
+    const {
+      eventTypeValues,
+      eventTopicValues,
+      form: { getFieldDecorator, getFieldValue }
+    } = this.props;
+
+    const urlType = getFieldValue("eventType");
+
+    return (
+      <Form className="main--eventForm" onSubmit={this.handleSubmit}>
+        <InputGroup size="large">
+          <InputAntd
+            withTip
+            label="Title"
+            tipInfo="Title for Event"
+            getFieldDecorator={getFieldDecorator}
+            name="title"
+            validationMsg="Please input your Event’s Title!"
+            placeholder="Event’s Title"
+            validation={{ max: 60 }}
+          />
+        </InputGroup>
+        <DropDownAntd
+          label="Event’s Type"
+          getFieldDecorator={getFieldDecorator}
+          name="eventType"
+          required
+          validationMsg="Please select your Event’s Type!"
+          placeholder="Event’s Type"
+          handleSelectChange={this.handleSelectChange}
+          optionsMenu={eventTypeValues}
+        />
+        <DropDownAntd
+          mode="multiple"
+          label="Event’s Topic"
+          getFieldDecorator={getFieldDecorator}
+          name="eventTopic"
+          required
+          validationMsg="Please select your Event Topic!"
+          placeholder="Event’s Topic"
+          handleSelectChange={this.handleSelectChange}
+          optionsMenu={eventTopicValues}
+        />
+        <TextAreaAntd
+          withTip
+          label="Description"
+          getFieldDecorator={getFieldDecorator}
+          name="description"
+          validationMsg="Please input your description!"
+          placeholder="Enter Event Description"
+          min={10}
+          max={false}
+        />
+        <Form.Item label={<span>Date and Time&nbsp;</span>}>
+          {getFieldDecorator("dateAndTime", {
+            rules: [
+              {
+                required: true,
+                message: "Please input your Date and Time!"
+              }
+            ]
+          })(
+            <DatePicker
+              style={{ width: "100%" }}
+              showTime
+              format="YYYY-MM-DD HH:mm:ss"
+              size="large"
+            />
+          )}
+        </Form.Item>
+        <InputGroup size="large">
+          <InputAntd
+            withTip={false}
+            label="Website"
+            getFieldDecorator={getFieldDecorator}
+            name="website"
+            validationMsg="Please input website!"
+            placeholder="Enter website"
+            validation={{
+              max: 60,
+              pattern: /^((https?|ftp|smtp):\/\/)?(www.)?[a-z0-9]+(\.[a-z]{2,}){1,3}(#?\/?[a-zA-Z0-9#]+)*\/?(\?[a-zA-Z0-9-_]+=[a-zA-Z0-9-%]+&?)?$/,
```

Comment:
@aminalakhsham If you think you might use this regex again in a different part of the app, it might be a good idea to pull it out into a file and import it into this component.